### PR TITLE
Implemented some libanki TODOs

### PR DESCRIPTION
**Pull request by RobozinhoD**
## Purpose / Description
I tried to find some TODO in the project and fixed these in libanki. I hope it helps

## Approach
Followed upstream's implementation

https://github.com/ankitects/anki/blob/edf59c2bb2a3dad36115af7518cdcbb3ee397089/pylib/anki/decks.py

## How Has This Been Tested?

I don't know how to test it

## Checklist
_Please, go through these checks before submitting the PR._

- [X] You have a descriptive commit message with a short title (first line, max 50 chars).
- [X] You have commented your code, particularly in hard-to-understand areas
- [X] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

**Review by BrayanDSO — APPROVED** [review]

File: AnkiDroid/src/main/java/com/ichi2/libanki/Decks.kt [hunk 1]
```diff
@@ -172,11 +181,10 @@ class Decks(private val col: Collection) {
         return col.backend.newDeck()
     }
 
-    @RustCleanup("implement and make public")
     @LibAnkiAlias("add_deck")
-    @Suppress("unused", "unused_parameter")
-    private fun addDeck(deck: Deck): OpChangesWithId {
-        TODO()
+    @Suppress("unused")
+    fun addDeck(deck: anki.decks.Deck): OpChangesWithId {
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/libanki/Decks.kt:186]: This is the only somewhat questionable one:

Do we want to continue using `Deck`, or move to the upstream Protobuf

For now, this is the easiest course of action

**Review by david-allison — APPROVED** [review]

**Comment by david-allison** [comment]
@RobozinhoD Thanks! Could you squash this and it'll be good to go

File: AnkiDroid/src/main/java/com/ichi2/libanki/Decks.kt (hunk 1 above)
Comment (RobozinhoD) [reply AnkiDroid/src/main/java/com/ichi2/libanki/Decks.kt:186]: I think that this one should be equal to libanki and that we can create another util or extension method for `Deck`

**Review by lukstbit — APPROVED** [review]
Ok!

I think we should leave these methods to be implemented when they are actually used in code. These PRs create unnecessary review requests for code that is not going to be used for the foreseeable future and they might even change by the time they are used and then instead of simply implementing the most recent version of the methods code we would need to debug to see why our stale implementation isn't working as expected.

**Pull request merged**